frontend/components/PageHeader/TeamsDropdownHeader.tsx

```diff
@@ -121,6 +123,7 @@ const TeamsDropdownHeader = ({
         onChange({ ...dropdownState, teamId: availableTeam?.id });
       }
     },
+    // TODO: add missing deps to this array if doens't cause bugs
```

Comment:
were we able to do this?

Reply:
it resulted in a bunch of other necessary changes that I haven't had the bandwidth to address. I'll look into it further today.